Sources/GRPCCore/GRPCClient.swift

```diff
@@ -0,0 +1,473 @@
+/// configuration.interceptors.add(StatsRecordingClientInterceptor())
+///
+/// // Override the timeout for the 'Get' method on the 'echo.Echo' service. This configuration
+/// // takes precedence over any set by the transport.
+/// let echoGet = MethodDescriptor(service: "echo.Echo", method: "Get")
+/// configuration.method.overrides[echoGet] = MethodConfiguration(
+///   executionPolicy: nil,
+///   timeout: .seconds(5)
+/// )
+///
+/// // Configure a fallback timeout for all RPCs if no configuration is provided in the overrides
+/// // or by the transport.
+/// let defaultMethodConfiguration = MethodConfiguration(executionPolicy: nil, timeout: seconds(10))
+/// configuration.method.defaults.setDefaultConfiguration(defaultMethodConfiguration)
+///
+/// // Finally create a transport and instantiate the client.
+/// let inProcessServerTransport = InProcessServerTransport()
+/// let inProcessClientTransport = InProcessClientTransport(serverTransport: inProcessServerTransport)
+/// let client = GRPCClient(transport: inProcessClientTransport, configuration: configuration)
+/// ```
+///
+/// ## Starting and stopping the client
+///
+/// Once you have configured the client, call ``run()`` to start it. Calling ``run()`` instructs the
+/// transport to start connecting to the server.
+///
+/// ```swift
+/// // Start running the client. 'run()' must be running while RPCs are execute so it's executed in
+/// // a task group.
+/// try await withThrowingTaskGroup(of: Void.self) { group in
+///   group.addTask {
+///     try await client.run()
+///   }
+///
+///   // Execute a request against the "echo.Echo" service.
+///   try await client.unary(
+///     request: ClientRequest.Single<[UInt8]>(message: [72, 101, 108, 108, 111, 33]),
+///     descriptor: MethodDescriptor(service: "echo.Echo", method: "Get"),
+///     serializer: IdentitySerializer(),
+///     deserializer: IdentityDeserializer(),
+///   ) { response in
+///     print(response.message)
+///   }
+///
+///   // The RPC has completed, close the client.
+///   client.close()
+/// }
+/// ```
+///
+/// The ``run()`` method won't return until the client has finished handling all requests. You can
+/// signal to the client that it should stop creating new request streams by calling ``close()``.
+/// This gives the client enough time to drain any requests already in flight. To stop the client
+/// more abruptly you can cancel the task running your client. If your application requires
+/// additional resources that need their lifecycles managed you should consider using [Swift Service
+/// Lifecycle](https://github.com/swift-server/swift-service-lifecycle).
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+public struct GRPCClient: Sendable {
+  /// The transport which provides a bidirectional communication channel with the server.
+  private let transport: any ClientTransport
+
+  /// The configuration used by the client.
+  public let configuration: Configuration
```

Comment:
Do we really want to make this public. Users cannot change anything anymore on it. Do you have a use-case in mind where people might want to access this?

Reply:
Visibility, i.e. being able to see what the configuration is.